tiledb/libtiledb.pyx

```diff
@@ -4484,6 +4484,33 @@ cdef class Array(object):
         Buffers will be used to satisfy the next index/query request.
         """
         self._buffers = buffers
+    
+    # pickling support: this is a lightweight pickle for distributed use.
+    #   simply treat as wrapper around URI, not actual data.
+    def __getstate__(self):
+        config_dict = self._ctx_().config().dict()
+        return (self.uri, self.mode, self.key, self.view_attr, self.timestamp_range, config_dict)
+
+    def __setstate__(self, state):
+        cdef:
+            unicode uri, mode
+            object view_attr = None
+            object timestamp_range = None
+            object key = None
+            dict config_dict = {}
+        uri, mode, key, view_attr, _timestamp_range, config_dict = state
+
+        if mode == 'r':
+            timestamp_range = _timestamp_range
```

Comment:
What is the reason for timestamp range only being preserved for reads?

Reply:
I think that was (my) oversight in the previous dense-only implementation. We should support for writes as well. cc @nguyenv